client/driver/docker.go

```diff
@@ -120,6 +120,13 @@ const (
 	// https://docs.docker.com/engine/reference/run/#block-io-bandwidth-blkio-constraint
 	dockerBasicCaps = "CHOWN,DAC_OVERRIDE,FSETID,FOWNER,MKNOD,NET_RAW,SETGID," +
 		"SETUID,SETFCAP,SETPCAP,NET_BIND_SERVICE,SYS_CHROOT,KILL,AUDIT_WRITE"
+
+	// This is cpu.cfs_period_us: the length of a period.
+	// The default values is 100 microsecnds represented in nano Seconds.
+	// Below is the documnentation:
+	// https://www.kernel.org/doc/Documentation/scheduler/sched-bwc.txt
+	// https://docs.docker.com/engine/admin/resource_constraints/#cpu
+	defaultCFSPeriod = 100000
```

Comment:
So for a task allocated 10% of the CPU it could pause for 90ms at a time? That seems awfully long, perhaps 10000 (10ms) or 1000 (1ms) increments would be a better level of granularity to ensure responsiveness for low priority interactive services?

Reply:
Yes, if a task is allocated 10% of CPU and we keep the default 100 microseconds cfs_peroid then it will be paused for 90 microseconds. We can change that to 10 ms, for that we will have to pass CPUPeriod as well (which would always be 10 ms )

Reply:
Side note: Sorry, I meant "ms" as in *milli* seconds. I know that kernel doc uses "ms" to mean microseconds, so I'm sorry for confusing the matter. Within Nomad code/comments/docs let's always use "ms" for millis and "us" or "μs" for micros.

I'd be in favor of lowering this to 10000μs (10ms) to minimize pause duration unless somebody has a strong preference.

Since it's only used for an internal calculation tweaking it in the future should be ok (and it should probably even be configurable once we make this cross-driver).

Reply:
@schmichael The default for cfs period in docker is 100000, changing this would mean people's experience of using the quota flag would differ between using normal docker cli and Nomad. If you want a lower value, maybe introduce another flag to tweak that but keep the default as is?

Reply:
@schmichael I am testing the changes that you recommended and running into an issue. When I specify cpu.cfs_period_us=10000 (milliseconds) and when I want to use 6% of CPU, I will have to specify cpu.cfs_quota_us=600(milliseconds). When I do that, it gives me an error: `CPU cfs quota cannot be less than 1ms (i.e. 1000)`.

Reply:
@diptanu has a good point. Let's match Docker's behavior and leave a comment to that affect.

Reply:
Although now I'm confused by the error @jaininshah9 mentioned. According to [Docker's API docs](https://docs.docker.com/engine/api/v1.35/#operation/ContainerCreate) this setting should be in *microseconds* (like the kernel itself expects), **not** nanoseconds as the code comment implies.

The Docker docs lead me to believe this should be the value?
 
```go
defaultCFSPeriod = 100
```

Reply:
@schmichael You may be right, that got me confused as well, I am digging into the error and will let you know.

Reply:
Okay, I  think initially I made a mistake mentioning that it's a nanosecond. The reason I mentioned nanosecond it is because the docker docs says (see screenshot below) the default value is 100 microseconds, which is wrong, it should say 100 milliseconds (which is what kernel doc says).
   
<img width="894" alt="screen shot 2018-02-06 at 6 53 23 pm" src="https://user-images.githubusercontent.com/7143248/35896086-56bb535e-0b6f-11e8-8004-40c32be11aeb.png">

I also tested and the docker API does not do any translation. Whatever value we pass in CPUPeriod is what we find under /sys/fs/cgroups...
**So the value is the code is correct, I need to update the comment on it though**

Sorry for the confusion I created. I will add a change which has a clear comment on what the defaultCFSPeriod refers to.